doc/7/getting-started/.react-native/screens/Chat/ChatClient.js

```diff
@@ -0,0 +1,122 @@
+        return false;
+    }
+    getMessage(hit, idx, array) {
+        // display will be set to true only if the previous message is from another day in goal to display only one time the dates
+        // and only the hours on each messages
+        let display = null;
+        // Not idx and array provided -> call from subscribe
+        let length = this.state.messages.length;
+        if (length == 0) {
+            display = false;
+        } else if (idx === null || array === null) {
+            display = this.displayDate( this.state.messages[length - 1].date, hit._source._kuzzle_info.createdAt)
+        } else { // idx and array provided -> call from fetch 
+            display = idx === 0? true : this.displayDate(array[idx - 1]._source._kuzzle_info.createdAt , hit._source._kuzzle_info.createdAt);
+        }
+        const message = {
+            // The unique id of the document containing the message
+            id: hit._id,
+            // The text of the message
+            message: hit._source.message,
+            // The creation date
+            date: hit._source._kuzzle_info.createdAt,
+            // The author name
+            author: hit._source.author,
+            // Boolean to display or not the date
+            displayDate: display
+        };
+        return message;
+    }
+    async fetchMessages() {
+        // Call the search method of the document controller
+        const results = await this.kuzzle.document.search(
+            "chat", // Name of the index
+            "messages", // Name of the collection
+            { sort: [{ ["_kuzzle_info.createdAt"]: { order: "desc" } }] }, // Query => Sort the messages by creation date
+            { size: 100 } // Options => get a maximum of 100 messages
+        );
+        // Add messages to our array after formating them
+        await this.setState({
+            messages: results.hits.reverse().map((hit, idx, array) => this.getMessage(hit, idx, array))
+        });
+    }
+
+    async subscribeMessages() {
+        // Call the subscribe method of the realtime controller and receive the roomId
+        const roomId = await this.kuzzle.realtime.subscribe(
+            "chat", // Id of the index
+            "messages", // Id of the collection
+            {}, // Filter
+            // Callback to receive notifications
+            notification => {
+                // Check if the notification interest us (only document creation)
+                if (notification.type !== "document") return;
+                if (notification.action !== "create") return;
+                // Add the new message to our array
+                this.setState({
+                    messages: [...this.state.messages.slice(), this.getMessage(notification.result, null, null)]
+                });
+            }
+        );
+        // Save the id of our subscription (we could need it to unsubscribe)
+        this.setState({ roomId: roomId });
+    }
+
+    async onSendMessage(text) {
```

Comment:
Nitpicking: poor variable name choice
It could be `message`, so you can also use shorthand property